src/utilities/index.ts

```diff
@@ -299,7 +299,10 @@ export const restoreOrder = ({
             }
           );
           if (!arrayItem) {
-            return;
+            return {
+              id: item.id,
+              ...(field.type === "blocks" && { blockType: item.blockType }),
+            };
           }
```

Comment:
This is the fix. All other changes in this pull request are related to writing tests in order to diagnose and fix the issue.